src/screens/groups/batch-payment-screen.tsx

```diff
@@ -0,0 +1,275 @@
+import Link from 'next/link'
+import React, { useState, useEffect, useContext } from 'react'
+import { Message, Grid, Header, Button } from 'semantic-ui-react'
+import { ErrorMessage, LoadingMessage } from '@/components'
+import { Web3Context } from '@/context/web3-context'
+import api, { APIError } from '@/lib/api'
+import type { GroupWithRecipients } from '@/pages/api/groups/[id]'
+import { useIsMobile } from '@/utils/use-is-mobile'
+import { BatchRecipientsList } from './components/recipients-list'
+import { BatchTransactionDialog } from './components/transaction-dialog'
+import { TransactionInfoDialog } from './components/transaction-info-dialog'
+import type { PaymentTransactionData } from './components/transaction-dialog/batch-transaction-dialog'
+
+interface Props {
+  groupId: string
+}
+export interface SelectedRecipients {
+  [key: string]: { selected: boolean; amount: number; address: string }
+}
+
+const Screen = ({ groupId }: Props) => {
+  const isMobile = useIsMobile()
+  const { connected } = useContext(Web3Context)
+
+  const [data, setData] = useState<GroupWithRecipients | undefined>(undefined)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [error, setError] = useState<string>('')
+
+  const [totalAmount, setTotalAmount] = useState<number>(0)
+  const [selectedRecipients, setSelectedRecipients] =
+    useState<SelectedRecipients>({})
+  const [transaction, setTransaction] = useState<string>('')
+  const [paymentTransactionData, setPaymentTransactionData] =
+    useState<PaymentTransactionData | null>(null)
+
+  const [transactionDialogOpen, setTransactionDialogOpen] =
+    useState<boolean>(false)
+  const [transactionInfoDialogOpen, setTransactionInfoDialogOpen] =
+    useState<boolean>(false)
+
+  const handleRowSelect = (id: string) => {
+    setSelectedRecipients((prevState) => ({
+      ...prevState,
+      [id]: {
+        selected: !prevState[id]?.selected,
+        amount: prevState[id]?.amount,
+        address: prevState[id]?.address,
+      },
+    }))
+  }
+
+  const handleSelectAll = () => {
+    const allIds = data?.recipients.map((recipient) => recipient.id)
+    const areAllSelected = allIds?.every(
+      (id) => selectedRecipients[id]?.selected
+    )
+
+    const newSelectedRecipients: SelectedRecipients = {}
+
+    allIds?.forEach((id) => {
+      const amount = selectedRecipients[id]?.amount
+      if (amount !== 0) {
```

Comment:
I think here is a bug, I need to test it carefully

Reply:
yeahh sure! why i have added that condition is - let's say there are x recipients and one of them has amount 0, so now if user click's on select all button that recipient who's having amount 0 will not get selected but his amount value was getting resetting to default. so now if user again click's on select all button then as that recipient has his amount value more than 0, he will also get selected, i thought this is a wrong behaviour that amount value should not get automatically reset from 0 to xx without user doing it. so now to fix that bug that condition helps, if user's amount value is 0 that will not get reset to default value.